crates/driver/openapi.yml

```diff
@@ -76,8 +154,107 @@ components:
           quote. For example, this Solver might not be aware of any liquidity between the tokens.
         type: object
         properties:
-          unfillable_reason:
+          unfillableReason:
             type: string
+    SolveRequest:
+      description: Request to the solve endpoint.
+      type: object
+      properties:
+        auction:
+          $ref: "#/components/schemas/Auction"
+        deadline:
+          description: |
+            The time until which the caller expects a response.
+
+            Encoded as ISO 8601 UTC.
+          type: string
+          example: "2020-12-03T18:35:18.814523Z"
+    SolveResponse:
+      description: Response of the solve endpoint.
+      type: object
+      properties:
+        objective:
+          description: |
+            The objective value of the solution.
+
+            `null` indicates that solving was successful but no solution was found.
+          type: number
+          nullable: true
+        signature:
+          description: |
+            Signature confirming that the Solver promised to have this solution for this auction.
+
+            TODO: Decide what is signed and how it is signed.
+          type: string
```

Comment:
I think to store the solver competition in a meaningful way it would be nice to expect the solvers to return a "plan" of their solution as well (i.e. prices, traded amounts, interactions plus inputs and outputs thereof, basically a settlement without call data). This would help us analyze solvers that are misbehaving and discard solutions which don't satisfy the properties we expect from a solution.
The `SettlementProposal` struct currently holds all the data which I thought would be required for that.

Reply:
Personally, I don't think we should store this. It can be derived by the token transfer amounts that they provide in their settlement.

I think a more appropriate additional inclusion (if we are worried about analyzing  settlements after the fact) would be a "this is the block I think the settlement is valid for" field.

This way, we should be able to derive interaction input and output amounts through EVM simulations and reduce the overall data stored, keeping the API leaner and easier to integrate.

Reply:
> Personally, I don't think we should store this.

Then why are we storing the whole solver competition right now?
I'm not concerned about analyzing the winning solution but the competition overall.

Reply:
Oh... right 🤦, we don't have calldata for solutions that don't win.

Then this is more interesting. Still don't know if we will need to validate runner-up solutions, but you are right that it might be interesting to store in the future.

Reply:
The main reason for the solver competition endpoint as it is now was just to provide some useful information about the competition for us and for other solver developers. Since we have this kind of info anyway we decided to store it. If we decided something like this here it would force all solvers to implement it which I'm not sure about. I can see arguments for and against it.

Reply:
Lets leave it out for now then, we can add it in if we see value once we co-locate.

Reply:
For me this question comes down to how much we trust the slashing mechanism. If everything goes smoothly solvers can just send us the objective value, win the competition, submit a solution and possibly get slashed for it.
If we think the slashing mechanism is robust and ergonomic enough to enforce a fair solver competition as we envision then we don't need the detailed solution before hand.
But I personally expect that it will take some time before the slashing mechanism is working as intended and we are happy with. Until then I see worth in automatically discarding solutions which appear to be unfair which requires this detailed information before hand.
I agree that we don't need it right now but I personally would at least like to see some concrete plans of how slashing is supposed to work before excluding this info from the API in good conscience because it would also vastly limit how much external parties can judge the fairness of the protocol.

Reply:
> For me this question comes down to how much we trust the slashing mechanism.

I'm not sure having this additional data would help though - just because a solver an just send a different transaction than what they advertised here. So ultimately, for the winning settlement, we will need to check the calldata and execution of the transaction regardless of whether or not this additional metadata gets included.

Reply:
That's true but I still think there is worth in having a higher hurdle to avoid falling back to the slashing mechanism as much as possible. Solvers could still execute different solutions than they promised but this way they would at least have to prove that they are able to come up with the best fair solution.
But again I agree that we shouldn't need the detailed execution plan when we trust in the slashing mechanism.

Reply:
For the solver competition visualisation, I would at least add information for which orders are settled and at what prices (so users can see what they would have gotten with the runner up etc). 

It might also be nice to report the expected gas amount (in which case the computation of the objective value could still happen in the autopilot and be potentially changed by us, e.g. when dropping gas cost from the objective value) without aligning all solvers to update their version at the same time.

Maybe we could also make it optional to submit call data (so that we can still show the solution simulations for some basic solvers and allow solvers to make use of our checks - some solvers are quite risk averse when it comes to trusting their own implementation)

Reply:
> so that we can still show the solution simulations for some basic solvers and allow solvers to make use of our checks

For this we provide the reference driver implementation that would simulate the calldata. I don't think we need to also submit it to the `autopilot` (which is what this API is documenting).